### ISSUE TITLE
2nd Subtitle Track name does not carry over from original to Encoded version.

### ISSUE THREAD

**Issue by matthoward01** [RandomEngy/VidCoder#1118]
### Problem Description

When writing to an mkv, the 2nd Subititle track name does not carry over to the final encoded mkv.  It is seeing the names correctly based on what the log says, but It seems the track name isn't making it into the "SubtitleList" for the 2nd track.  I have not tested this for more then two subtitle tracks.  I have included the subtitle parts of the log.

### What version of VidCoder are you running?

8.24

### Encode Log

```text
Stream #0:3(eng): Subtitle: ass (default)
    Metadata:
      title           : Signs & Songs@EMBER
      BPS             : 12313
      DURATION        : 00:18:49.270000000
      NUMBER_OF_FRAMES: 6672
      NUMBER_OF_BYTES : 1738174
      _STATISTICS_WRITING_APP: mkvmerge v65.0.0 ('Too Much') 64-bit
      _STATISTICS_WRITING_DATE_UTC: 2022-07-08 14:18:07
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
  Stream #0:4(eng): Subtitle: ass
    Metadata:
      title           : Dialogue@9volt
      BPS             : 12087
      DURATION        : 00:23:13.040000000
      NUMBER_OF_FRAMES: 9016
      NUMBER_OF_BYTES : 2104729
      _STATISTICS_WRITING_APP: mkvmerge v65.0.0 ('Too Much') 64-bit
      _STATISTICS_WRITING_DATE_UTC: 2022-07-08 14:18:07
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES

"SubtitleList": [
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 1,
        "Offset": 0,
        "Track": 0,
        "Name": "Signs \u0026 Songs@EMBER"
      },
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 2,
        "Offset": 0,
        "Track": 1
      }
    ]
```

**Comment by matthoward01** [comment]
I looked at the source code. It took me a couple hours, but I think I found the issue.
In the method "**CreateSubtitles**" of "**JsonEncodeFactory.cs**", on line 881 the **sourceSubtitle.TrackNumber**  should be **<=** to the **title.SubtitleList.Count** instead of **<**.
Since the **sourceSubtitle.TrackNumber** is starting at 1 instead of 0, the part where you get the name from the **sourceSubtitleTrack.Name** never happens.  Below is what made it work for me.  All i know is this worked for the files I was working with. I don't know if it has any further implications, but in all my tests, it worked as I expected.
```
if (sourceSubtitle.TrackNumber > 0 && sourceSubtitle.TrackNumber <= title.SubtitleList.Count)
{
        SourceSubtitleTrack sourceSubtitleTrack = title.SubtitleList[sourceSubtitle.TrackNumber - 1];
        forcedOnly = forcedOnly && HandBrakeEncoderHelpers.SubtitleCanSetForcedOnly(sourceSubtitleTrack.Source);

       if (string.IsNullOrEmpty(name))
       {
              name = sourceSubtitleTrack.Name;
       }
}
```

**Comment by RandomEngy** [comment]
Good catch. I'll get this in an official fix in a couple weeks when back from vacation.

**Comment by RandomEngy** [comment]
Fixed in 8.25.
